apps/hrms/views.py

```diff
@@ -404,45 +262,247 @@ def update(self, request, *args, **kwargs):
+            return result if result else self.retrieve(self, request, *args, **kwargs)
+        try:
+            logger.info("Retrieving all Employees")
+
+            page = int(request.query_params.get('page', 1))  # Default to page 1 if not provided
+            limit = int(request.query_params.get('limit', 10)) 
+
+            queryset = Employees.objects.all()
+
+            # Apply filters manually
+            if request.query_params:
+                filterset = EmployeesFilter(request.GET, queryset=queryset)
+                if filterset.is_valid():
+                    queryset = filterset.qs 
+
+            total_count = Employees.objects.count()
+
+            serializer = EmployeesSerializer(queryset, many=True)
+            logger.info("Employees data retrieved successfully.")
+            # return build_response(queryset.count(), "Success", serializer.data, status.HTTP_200_OK)
+            return filter_response(queryset.count(),"Success",serializer.data,page,limit,total_count,status.HTTP_200_OK)
+
+        except Exception as e:
+            logger.error(f"An unexpected error occurred: {str(e)}")
+            return build_response(0, "An error occurred", [], status.HTTP_500_INTERNAL_SERVER_ERROR)
+        
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Retrieves a Employees and its related data (EmployeeDetails).
+        """
+        try:
+            pk = kwargs.get('pk')
+            if not pk:
+                logger.error("Primary key not provided in request.")
+                return build_response(0, "Primary key not provided", [], status.HTTP_400_BAD_REQUEST)
+            
+            # Retrieve the Employees instance
+            employee = get_object_or_404(Employees, pk=pk)
+            employee_serializer = EmployeesSerializer(employee)
+
+            # Retrieve EmployeeDetails data
+            employee_details_data = self.get_related_data(EmployeeDetails, EmployeeDetailsSerializer, 'employee_id', pk)
+            emp_details_data = employee_details_data[0] if employee_details_data else {}
+
+            # Customizing the response data
+            custom_data = {
+                "employee": employee_serializer.data,
+                "employee_details": emp_details_data
+            }
+            logger.info("Employee and related data retrieved successfully.")
+            return build_response(1, "Success", custom_data, status.HTTP_200_OK)
+
+        except Http404:
+            logger.error("Employee with pk %s does not exist.", pk)
+            return build_response(0, "Record does not exist", [], status.HTTP_404_NOT_FOUND)
+        except Exception as e:
+            logger.exception("An error occurred while retrieving Employees with pk %s: %s", pk, str(e))
+            return build_response(0, "An error occurred", [], status.HTTP_500_INTERNAL_SERVER_ERROR)
+        
+    def get_related_data(self, model, serializer_class, filter_field, filter_value):
```

Comment:
**get_related_data** you can import this function from **utils_methods.py** 
please remove here and do import.